src/votekit/elections/election_types.py

```diff
@@ -1117,3 +1118,183 @@ def __init__(
+            if not self.next_round():
+                self.state = ElectionState(
+                    curr_round=self.state.curr_round,
+                    elected=elected,
+                    eliminated_cands=remaining,
+                    remaining=[],
+                    profile=new_profile,
+                    previous=self.state.previous,
+                )
+            return self.state
+
+    def run_election(self):
+        # run steps until we elect the required number of candidates
+        while self.next_round():
+            self.run_step()
+
+        return self.state
+
+
+class BoostedRandomDictator(RandomDictator):
+    """
+    Modified random dictator where we
+        - Choose a winner randomly from the distribution of first
+          place votes with probability (1 - 1/(# Candidates - 1))
+        - Choose a winner via a proportional to squares rule with
+          probability 1/(# of Candidates - 1)
+
+    For multi-winner elections
+    repeat this process for every winner, removing that candidate from every voter's ballot
+    once they have been elected
+
+    Args:
+      profile (PreferenceProfile): PreferenceProfile to run election on
+      seats (int): number of seats to select
+
+    Attributes:
+      _profile (PreferenceProfile): PreferenceProfile to run election on
+      seats (int): Number of seats to be elected.
+
+    """
+
+    def __init__(self, profile: PreferenceProfile, seats: int):
+        # the super method says call the Election class
+        # ballot_ties = True means it will resolve any ties in our ballots
+        super().__init__(profile, seats)
+
+    def run_step(self):
+        if self.next_round():
+            remaining = self.state.profile.get_candidates()
+            u = random.uniform(0, 1)
+
+            if len(remaining) == 1:
+                winning_candidate = remaining[0]
+
+            elif u <= 1 / (len(remaining) - 1):
+                # Choose via proportional to squares
+                candidate_votes = {c: 0 for c in remaining}
+                for ballot in self.state.profile.get_ballots():
+                    top_choice = list(ballot.ranking[0])[0]
+                    candidate_votes[top_choice] += float(ballot.weight)
+
+                squares = np.array(
+                    [(i / len(remaining)) ** 2 for i in candidate_votes.values()]
+                )
+                sum_of_squares = np.sum(squares)
+                probabilities = squares / sum_of_squares
+                winning_candidate = np.random.choice(remaining, p=probabilities)
+
+            else:
+                ballots = self.state.profile.ballots
+                weights = [b.weight for b in ballots]
+                random_ballot = random.choices(
+                    self.state.profile.ballots, weights=weights, k=1
+                )[0]
+                # randomly choose a winner according to first place rankings
+                winning_candidate = list(random_ballot.ranking[0])[0]
```

Comment:
random.choice(random_ballot.ranking[0])[0]

Reply:
After some googling I believe list does not randomize the order of the set. Unfortunately random.choice() doesn't work on sets either. It could be slow but a good option might be: random.choice(list(random_ballot.ranking[0])). What do you think?

Reply:
So the order of the set is determined by the hash of the underlying type. There is an environment variable, PYTHONHASHSEED, that controls this for security reasons. If you want this to be deterministic, you will need to use something other than a set. You can make use of the OrderedSet class in the collections module, or just use a dictionary. Convertiing to a list is also fine here. It doesn't really take up a lot of overhead in the grand scheme of things.